reyna/src/com/b2msolutions/reyna/services/StoreService.java

```diff
@@ -38,28 +40,56 @@ public static void setCellularDataBlackout(Context context, TimeRange range) {
         new Preferences(context).saveCellularDataBlackout(range);
     }
 
-	@Override
-	protected void onHandleIntent(Intent intent) {
+    public static void setStorageSizeLimit(Context context, long limit) {
```

Comment:
this check is done inside the shrink operation itself. I put it there because it uses repository specific operations like getDbSize and SIZE_DIFFERENCE_TO_START_CLEANING constant (which is 300Kb) which is repo. internal information.
@ivanbokii